### ISSUE TITLE
Cloud save not showing

### ISSUE THREAD

**Issue by Ghostkwebb** [timkurvers/valheim-macos#25]
So I was using parallels cause Valheim mac wasn't working then it worked and I saw that the cloud saves weren't showing up.
What do you think I should do to fix this?

**Comment by Ghostkwebb** [comment]
I'm playing on a vm called parallels and I'm pretty sure these are all cloud saves 
<img width="1728" alt="image" src="https://github.com/timkurvers/valheim-macos/assets/61558762/b2d81681-bec0-488f-aa0f-2bc5a121022f">

**Comment by Ghostkwebb** [comment]
![Uploading image.png…]()

**Comment by Ghostkwebb** [comment]
<img width="986" alt="image" src="https://github.com/timkurvers/valheim-macos/assets/61558762/f7128f23-54fe-4ce1-aec5-967083f664d6">

**Comment by timkurvers** [comment]
If this is an issue with the Windows version of the game you should open a support ticket with Iron Gate.

If this is while using the unofficial macOS build, have a look at `~/Library/Application Support/Steam/logs/cloud_log.txt` first, to see if there's trouble with Steam syncing content.

**Comment by Ghostkwebb** [comment]
So it is with the unofficial build. I was referring to how it saves on Windows but not on the unofficial build. Also here is the cloud_log.
[cloud_log.txt](https://github.com/timkurvers/valheim-macos/files/11598101/cloud_log.txt)

This is a pic of the unofficial build.
<img width="1728" alt="image" src="https://github.com/timkurvers/valheim-macos/assets/61558762/adcab371-f146-4e8c-991a-a4931402dc5c">

The above pic was the windows VM version

**Comment by timkurvers** [comment]
That does indeed look like Steam did not manage to sync your files properly:

```
[2023-05-27 21:09:19] [AppID 892970] characters/ghostkwebb.fch has local changes but mismatch - creating a conflict. Local/Remote SHA: ...
[2023-05-27 21:09:19] [AppID 892970] characters/ghostkwebb.fch.old has local changes but mismatch - creating a conflict. Local/Remote SHA: ...
[2023-05-27 21:09:19] [AppID 892970] serverlist/recent has local changes but mismatch - creating a conflict. Local/Remote SHA: ...
[2023-05-27 21:09:19] [AppID 892970] serverlist/recent.old has local changes but mismatch - creating a conflict. Local/Remote SHA: ...
[2023-05-27 21:09:19] [AppID 892970] Sync conflict for file characters/ghostkwebb.fch
[2023-05-27 21:09:19] [AppID 892970] Sync conflict for file characters/ghostkwebb.fch.old
[2023-05-27 21:09:19] [AppID 892970] Sync conflict for file serverlist/recent
[2023-05-27 21:09:19] [AppID 892970] Sync conflict for file serverlist/recent.old
[2023-05-27 21:09:19] [AppID 892970] Upload failed due to conflicts in build list
```

Hard to say, but it seemed to work up until May 25th, then stopped working the 27th onwards. This could be tricky to resolve as Steam will not allow you to resolve conflicts through the UI since macOS is not officially supported for Valheim.

To discard any local changes made in the local macOS version, and start using your cloud saves, you could try the following:

1. Ensure Valheim and Steam (macOS) are _not_ running
2. Make a backup of your local Valheim user data, located in `~/Library/Application Support/Steam/userdata/<steam user id>/892970`
3. Launch Steam (macOS), make sure you are logged in / online
4. Launch Valheim custom build

All of this is highly experimental, so make sure you have backups.

**Comment by wujekbogdan** [comment]
I'm having exactly the same issue. I can see my cloud saves in the `~/Library/Application Support/Steam/userdata/<steam user id>/892970/remote` directory. I can browse cloud saves via "Manage Saves" UI, but neither worlds nor characters appear in the game menu. The list is blank. I followed the instructions from [this post](https://github.com/timkurvers/valheim-macos/issues/25#issuecomment-1574367305), but it didn't help.

**Comment by cparty14** [comment]
Seems it's a common problem because I don't see any of my characters/worlds stored in my cloud save.
After game launch there's notification about the problem and cloud data won't be available.

**Comment by ShotgunF15** [comment]
I installed a few weeks ago and having similar problems... runs great with new start but if I try to use anything from the cloud it will not work.  It 'sees' characters when I view what is in the cloud but when I go to start a game, it only shows the new locally created characters.  Added to this, I don't even see any of the save architecture on my drive.  There is no:  ~/Library/Application Support/Steam/userdata/<steam user id>/892970/remote folder on my computer.   The local folder doesn't appear either however the fact that I can play with newly created characters means they are being saved somewhere!
Any ideas would be greatly appreciated

**Comment by timkurvers** [comment]
> The local folder doesn't appear either however the fact that I can play with newly created characters means they are being saved somewhere! Any ideas would be greatly appreciated

Could you check `~/Library/Application Support/unity.IronGate.Valheim-macOS-Custom/` and see if you recognize your local characters/worlds?

**Comment by ShotgunF15** [comment]
There is no unity.Irongate.Valheim…. Folder when I look in~/Library/ApplicationSupport/  I feel I must be doing something wrong but game works great, albeit only with new characters.  It ‘sees’ my cloud characters but will not let me use them or copy to local (show them as 0kb)Sent from my iPhoneOn Jul 22, 2023, at 16:20, Tim Kurvers ***@***.***> wrote:﻿

The local folder doesn't appear either however the fact that I can play with newly created characters means they are being saved somewhere! Any ideas would be greatly appreciated

Could you check ~/Library/Application Support/unity.IronGate.Valheim-macOS-Custom/ and see if you recognize your local characters/worlds?

**Comment by wujekbogdan** [comment]
> There is no unity.Irongate.Valheim…. Folder when I look in~/Library/ApplicationSupport/ 

It's the `~/Library/Application Support/` directory not the `~/Library/ApplicationSupport/` you're looking for.

**Comment by wujekbogdan** [comment]
> The second time I loaded the game up, the characters came up fine...

Sadly I can't confirm that. My saves won't load no matter how many times I try.

**Comment by timkurvers** [comment]
Closing this issue as the official macOS client is now available! 🥳 (see #104)
